src/components/ScrollView/PagingScrollViewBase.js

```diff
@@ -0,0 +1,224 @@
+import React, {Component} from 'react';
+import View from '../View';
+import Animated from '../../apis/Animated';
+import Easing from 'animated/lib/Easing';
+
+const normalizeScrollEvent = (parent, xOffset) => ({
+    nativeEvent: {
+        contentOffset: {
+            get x() {
+                return xOffset;
+            },
+            get y() {
+                return 0;
+            }
+        },
+        contentSize: {
+            get height() {
+                return parent._contentHeight;
+            },
+            get width() {
+                return parent._contentWidth;
+            }
+        },
+        layoutMeasurement: {
+            get height() {
+                return parent._parentHeight;
+            },
+            get width() {
+                return parent._parentWidth;
+            }
+        }
+    },
+    timeStamp: Date.now()
+});
+
+export default class PagingScrollViewBase extends Component {
```

Comment:
Because it does not really use the base at all. Scroll events are raised using onTouchMove, you can snap browser scrolling. Switching the base seemed like a good idea to me. I'm open to ideas here since you know the code better